docs/source/error_code_list2.rst

```diff
@@ -231,6 +231,40 @@ incorrect control flow or conditional checks that are accidentally always true o
         # Error: Statement is unreachable  [unreachable]
         print('unreachable')
 
+.. _code-deprecated:
+
+Check that imported or used feature is deprecated [deprecated]
+--------------------------------------------------------------
+
+If you use :option:`--warn-deprecated <mypy --warn-deprecated>`, mypy generates a note if
```

Comment:
It looks like there has been some back and forth so docs are out of sync now. To summarize how I see the logic should be:
* Deprecation messages are always shown as notes by default (with code shown, see my previous comment), this will result in exit code 0 (which is kind of my main concern, I don't want users CI to break unnecessarily on new mypy version).
* People who still don't want to see the notes, can use `--disable-error-code=deprecated` and/or `# type: ignore[deprecated]`.
* There should be a flag that *turns notes into errors* (for more pedantic people). They will still be able to selectively silence the errors using `# type: ignore[deprecated]`.

Reply:
> * There should be a flag that _turns notes into errors_ (for more pedantic people).

Does Mypy implement other checks that can result in errors or notes, depending on an option? (I'm just asking to avoid reinventing the wheel.)

Reply:
TBH I don't remember. But in any case it should be as simple as changing last line in your `warn_deprecated()` method in `checker.py` to something like
```python
if self.options.error_on_deprecated:  # or other flag name
    self.error(...)
else:
    self.note(...)
```

Reply:
That seems doable, thanks.  I will implement it later.

Reply:
Done!  `report-deprecated-as-error` is the shortest understandable description I could come up with.  Other suggestions are welcome, of course.